### PR TITLE
CP-32106: backport of CA-323765

### PR DESCRIPTION
**Pull request by lippirk**
Signed-off-by: lippirk <ben.anson@citrix.com>

### CODE REVIEW

**Comment by edwintorok** [comment]
Can you point to the commit hash of the commit that you have backported?

**Comment by lippirk** [comment]
> Can you point to the commit hash of the commit that you have backported?

c9d61af324e957ba2486bf133742a0e69703cd83

File: lib/network_utils.ml [hunk 1]
```diff
@@ -896,7 +916,15 @@ module Ovs = struct
 		in
 		let vif_arg =
 			let existing_vifs = List.filter (fun iface -> not (Sysfs.is_physical iface)) (bridge_to_interfaces name) in
-			List.flatten (List.map (fun vif -> ["--"; "--may-exist"; "add-port"; name; vif]) existing_vifs)
+        let ifaces_with_type =
+          let raw = vsctl ~log:false ["--bare"; "-f"; "table"; "--"; "--columns=name,type"; "find"; "interface"; {|type!=\"\"|}] in
+          let lines = String.trim raw |> fun x -> Stringext.split x ~on:'\n' |> List.filter (fun x -> x <> "") in
```
Comment (psafont) [inline lib/network_utils.ml]: This looks like a faithful reimplementation of `Astring.cuts` using stdext's 2.1.0 version.

**Comment by lindig** [comment]
The failure of Travis CI can be ignored. This was tested in a proper environment.

File: lib/network_utils.ml [hunk 2]
```diff
@@ -37,6 +37,20 @@ let bonding_dir = "/proc/net/bonding/"
 let fcoedriver = ref "/opt/xensource/libexec/fcoe_driver"
 let mac_table_size = ref 10000
 
+let filter_map f =
```
Comment (psafont) [inline lib/network_utils.ml]: is there any reason to not use stdext's `Listext.filter_map` ?

**Comment by lindig** [comment]
Please update the commit message to include the commit hash of the original.

File: lib/network_utils.ml (hunk 2 above)
Comment (lippirk) [reply lib/network_utils.ml]: i did try and add a dependency on `xapi-stdext-std` in the oasis file, but got an error

in the container you get 
```
[planex:networkdbackport local]$ ocamlfind list | grep stdex
stdext              (version: 2.1.0)
```

which made me think that it is not available

Comment (lippirk) [reply lib/network_utils.ml]: mb, `Listext` is available and gets opened so we can simply use `List.filter_map`

**Comment by lindig** [comment]
Looks good to me. Merge this and after a build succeeds tag it?

**Comment by lippirk** [comment]
> Looks good to me. Merge this and after a build succeeds tag it?

will do - thanks!

**Pull request merged**